docs/migration.md

```diff
@@ -0,0 +1,195 @@
+# Prometheus 2.0 Migration Guide.
```

Comment:
Our docs style is "no title-casing" throughout, so "Prometheus 2.0 migration guide", similar for headlines below.